solutions/page_object/pages/base_page.py

```diff
@@ -0,0 +1,36 @@
+import math
+import time
+
+from selenium.common.exceptions import NoAlertPresentException, NoSuchElementException
+
+
+class BasePage:
+    def __init__(self, browser, url, timeout=10):
+        self.browser = browser
+        self.url = url
+        self.browser.implicitly_wait(timeout)
+
+    def open(self):
+        self.browser.get(self.url)
+
+    def is_element_present(self, how, what):
+        try:
+            self.browser.find_element(how, what)
+        except NoSuchElementException:
+            return False
+        return True
+
+    def solve_quiz_and_get_code(self):
+        alert = self.browser.switch_to.alert
+        x = alert.text.split(" ")[2]
```

Comment:
Понятно, что скопировано, но это здесь явно лишнее.

```suggestion
        x = alert.text.split()[2]
```